lib/matplotlib/style/core.py

```diff
@@ -84,6 +91,101 @@ def use(style):
                 raise IOError(msg % style)
 
 
+def get_parents_styles(styles):
+    """ Returns a list of all styles with parent styles included
+    Parameters
+    ----------
+    styles = list of str and/or dict
+    Example
+    -------
+    Let us consider multiple stylesheet calling each other
+    main
+    |
+    |- a
+    |  |- a1
+    |  |- a2
+    |
+    |- b
+    |  |- b1
+    |  |- b2
+    |
+    |- c
+       |- c1
+       |- c2
+    This means that `main` calls `a`, `b`, and `c`
+    Similar pattern for a, b and c.
+    When calling `matplotlib.style.use(['main']), we will call all the
+    stylesheets recursively add apply all styles. In this example, it would
+    return the following list
+    ['c2', 'c1', 'c', 'b2', 'b1', 'b', 'a2', 'a1', 'a', 'main']
+    """
+    styles = [styles]
+    has_parent = True
+    full_styles = []
+    temp_styles = styles[:]
+
+    while has_parent:
+        has_parent = False
+        for style in styles:
+            if style in full_styles:
+                continue
+            full_styles.append(style)
+
+            if _is_dict_like(style):
+                continue
+
+            style_dic = get_style_parameters(style)
+
+            if style_dic.get('style', None):
+                has_parent = True
+                parent = style_dic['style']
+                temp_styles.insert(temp_styles.index(style)+1, parent)
+                temp_styles = list(_list_flatten(temp_styles))
+
+        styles = temp_styles[:]
+    styles = styles[::-1]
+    return styles
+
+
+def _is_dict_like(style):
```

Comment:
If there's not an implementation in Matplotlib, I think the one in the link above is a bit clearer.